src/Sulu/Bundle/AdminBundle/Resources/js/components/Popover/Popover.js

```diff
@@ -0,0 +1,148 @@
+import {action, computed, observable} from 'mobx';
+import type {Node, ElementRef} from 'react';
+import Backdrop from '../Backdrop';
+import type {PopoverDimensions} from './types';
+import PopoverPositioner from './PopoverPositioner';
+import popoverStyles from './popover.scss';
+
+type Props = {
+    open: boolean,
+    children?: (setPopoverElementRef: (ref: ElementRef<*>) => void, style: Object) => Node,
+    onClose?: () => void,
+    /** The element which will be used to position the popover */
+    anchorElement: ElementRef<*>,
+    centerChildElement?: ElementRef<*>,
+    horizontalOffset: number,
+    verticalOffset: number,
+};
+
+@observer
+export default class Popover extends React.PureComponent<Props> {
+    static defaultProps = {
+        open: false,
+        horizontalOffset: 0,
+        verticalOffset: 0,
+    };
+
+    @observable popoverRef: ElementRef<'div'>;
+
+    @observable scrollHeight: number;
+
+    @observable scrollWidth: number;
+
+    scrollTop: number;
+
+    componentDidMount() {
+        window.addEventListener('blur', this.close);
+        window.addEventListener('resize', this.close);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('blur', this.close);
+        window.removeEventListener('resize', this.close);
+    }
+
+    componentDidUpdate() {
+        if (this.popoverRef) {
+            this.popoverRef.scrollTop = this.scrollTop || 0;
+        }
+    }
+
+    close = () => {
+        if (this.props.onClose) {
+            this.props.onClose();
+        }
+    };
+
+    @computed get dimensions(): PopoverDimensions {
+        const {
+            anchorElement,
+            verticalOffset,
+            horizontalOffset,
+            centerChildElement,
+        } = this.props;
+        const {
+            top = 0,
+            left = 0,
+            width = 0,
+            height = 0,
+        } = anchorElement.getBoundingClientRect();
+        const centerChildOffsetTop = (centerChildElement) ? centerChildElement.offsetTop : 0;
+        const alignOnVerticalAnchorEdges = (!centerChildElement) ? true : false;
+
+        return PopoverPositioner.getCroppedDimensions(
+            this.scrollWidth,
+            this.scrollHeight,
+            top,
+            left,
+            width,
+            height,
+            horizontalOffset,
+            verticalOffset,
+            centerChildOffsetTop,
+            alignOnVerticalAnchorEdges,
+        );
+    }
+
+    updateDimensions() {
+        const {
+            scrollWidth,
+            scrollHeight,
+        } = this.popoverRef;
+
+        this.setScrollDimensions(
+            scrollWidth,
+            scrollHeight,
+        );
+    }
+
+    @action setScrollDimensions(scrollWidth: number, scrollHeight: number) {
+        this.scrollWidth = scrollWidth;
+        this.scrollHeight = scrollHeight;
+    }
+
+    handleBackdropClick = this.close;
+
+    @action setPopoverRef = (popoverRef: ElementRef<*>) => {
```

Comment:
I think the name of this function is confusing... The ref is not from the popover, isn't it? How about `setPopoverChildRef`?